### PR TITLE
Fixes #26323 - Remove max_tasks_per_child setting

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by chris1984** [comment]
@ekohl let me know if that looks right, it sounds right but only the 2nd time I am doing one of these.

**Review by ekohl — APPROVED** [review]
LGTM. Can be merged after the puppet module PRs have been merged.

**Pull request merged**
